app/src/main/java/com/valhalla/thor/presentation/utils/AppIconLoader.kt

```
@@ -71,30 +88,41 @@ class AppIconFetcher(
+            // Report sampled only when the rendered bitmap is smaller than the drawable's
+            // intrinsic size, so a full-size render can be reused for larger targets.
+            val sampled = bitmap.width < drawable.intrinsicWidth ||
+                bitmap.height < drawable.intrinsicHeight
             ImageFetchResult(
-                image = drawable.asImage(),
-                isSampled = false,
+                image = BitmapDrawable(context.resources, bitmap).asImage(),
+                isSampled = sampled,
                 dataSource = DataSource.DISK
             )
         } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)